src/sdk/pynni/nni/bohb_advisor/bohb_advisor.py

```diff
@@ -0,0 +1,594 @@
+
+        Parameters
+        ----------
+        data: JSON object
+            search space of this experiment
+
+        Returns
+        -------
+        ConfigSpace:
+            search space in ConfigSpace format
+        """
+        search_space = data
+        cs = CS.ConfigurationSpace()
+        for var in search_space:
+            _type = str(search_space[var]["_type"])
+            if _type == 'choice':
+                cs.add_hyperparameter(CSH.CategoricalHyperparameter(
+                    var, choices=search_space[var]["_value"]))
+            elif _type == 'randint':
+                cs.add_hyperparameter(CSH.UniformIntegerHyperparameter(
+                    var, lower=0, upper=search_space[var]["_value"][0]))
+            elif _type == 'uniform':
+                cs.add_hyperparameter(CSH.UniformFloatHyperparameter(
+                    var, lower=search_space[var]["_value"][0], upper=search_space[var]["_value"][1]))
+            elif _type == 'quniform':
+                cs.add_hyperparameter(CSH.UniformFloatHyperparameter(
+                    var, lower=search_space[var]["_value"][0], upper=search_space[var]["_value"][1], 
+                    q=search_space[var]["_value"][2]))
+            elif _type == 'loguniform':
+                cs.add_hyperparameter(CSH.UniformFloatHyperparameter(
+                    var, lower=search_space[var]["_value"][0], upper=search_space[var]["_value"][1], 
+                    log=True))
+            elif _type == 'qloguniform':
+                cs.add_hyperparameter(CSH.UniformFloatHyperparameter(
+                    var, lower=search_space[var]["_value"][0], upper=search_space[var]["_value"][1], 
+                    q=search_space[var]["_value"][2], log=True))
+            elif _type == 'normal':
+                cs.add_hyperparameter(CSH.NormalFloatHyperparameter(
+                    var, mu=search_space[var]["_value"][1], sigma=search_space[var]["_value"][2]))
+            elif _type == 'qnormal':
+                cs.add_hyperparameter(CSH.NormalFloatHyperparameter(
+                    var, mu=search_space[var]["_value"][1], sigma=search_space[var]["_value"][2], 
+                    q=search_space[var]["_value"][3]))
+            elif _type == 'lognormal':
+                cs.add_hyperparameter(CSH.NormalFloatHyperparameter(
+                    var, mu=search_space[var]["_value"][1], sigma=search_space[var]["_value"][2], 
+                    log=True))
+            elif _type == 'qlognormal':
+                cs.add_hyperparameter(CSH.NormalFloatHyperparameter(
+                    var, mu=search_space[var]["_value"][1], sigma=search_space[var]["_value"][2], 
+                    q=search_space[var]["_value"][3], log=True))
+            else:
+                raise ValueError(
+                    'unrecognized type in search_space, type is {}'.format(_type))
+
+        self.search_space = cs
+        return True
```

Comment:
return True? But in the docstring...

        Returns
        -------
        ConfigSpace:
            search space in ConfigSpace format
        """

Reply:
Modified.